src/transports/ws/sws.c

```diff
@@ -70,9 +71,9 @@
 #define NN_WS_OPCODE_UNUSED5 0x05
 #define NN_WS_OPCODE_UNUSED6 0x06
 #define NN_WS_OPCODE_UNUSED7 0x07
-#define NN_WS_OPCODE_CLOSE NN_WS_MSG_TYPE_CLOSE
-#define NN_WS_OPCODE_PING NN_WS_MSG_TYPE_PING
-#define NN_WS_OPCODE_PONG NN_WS_MSG_TYPE_PONG
+#define NN_WS_OPCODE_CLOSE 0x08
+#define NN_WS_OPCODE_PING 0x09
+#define NN_WS_OPCODE_PONG 0x0A
```

Comment:
At the time this was developed, it was not explicitly stated whether a browser or user agent could unsolicitedly (from the perspective of app code) could PING. For this reason, I chose to defensively go ahead and aim for full RFC 6455 compliance, such that your-browser-du-jour would not tend to crash `libnanomsg`. Now, in 2015, it's still not any more clear. With hindsight, I would tend to criticize the existence of PING/PONG in RFC 6455, since it just kinda throws a monkey wrench into things.